system-x/services/samba/src/main/java/software/tnb/samba/validation/SambaValidation.java

```diff
@@ -0,0 +1,122 @@
+
+            mkdirs(share, file.getParent());
+
+            write(share, file.getPath(), content);
+        }
+    }
+
+    public String read(File file) throws IOException {
+        try (Connection connection = sambaClient.connect(sambaServer.host(), sambaServer.port())) {
+            Session session = connection.authenticate(
+                new com.hierynomus.smbj.auth.AuthenticationContext(
+                    sambaServer.account().user(),
+                    sambaServer.account().password().toCharArray(),
+                    null
+                )
+            );
+
+            DiskShare share = (DiskShare) session.connectShare(sambaServer.shareName());
+            return read(share, file.getPath());
+        }
+    }
+
+    private String read(DiskShare share, String filePath) throws IOException {
+        if (!share.fileExists(filePath)) {
+            throw new IllegalStateException("File not found: " + filePath);
+        }
+
+        try (com.hierynomus.smbj.share.File file = share.openFile(
+            filePath,
+            EnumSet.of(AccessMask.GENERIC_READ),
+            EnumSet.of(FileAttributes.FILE_ATTRIBUTE_NORMAL),
+            EnumSet.of(SMB2ShareAccess.FILE_SHARE_READ),
+            SMB2CreateDisposition.FILE_OPEN,
+            EnumSet.noneOf(SMB2CreateOptions.class)
+        ); InputStream inputStream = file.getInputStream()) {
+            return new String(inputStream.readAllBytes(), StandardCharsets.UTF_8);
+        }
+    }
+
+    private void write(DiskShare share, String fileName, String content) throws IOException {
+        if (!share.fileExists(fileName)) {
+            try (com.hierynomus.smbj.share.File file = share.openFile(
+                fileName,
+                EnumSet.of(AccessMask.GENERIC_WRITE),
+                EnumSet.of(FileAttributes.FILE_ATTRIBUTE_NORMAL),
+                SMB2ShareAccess.ALL,
+                SMB2CreateDisposition.FILE_CREATE,
+                EnumSet.of(SMB2CreateOptions.FILE_NON_DIRECTORY_FILE)
+            )) {
+                try (OutputStream outputStream = file.getOutputStream()) {
+                    outputStream.write(content.getBytes(StandardCharsets.UTF_8));
+                }
+            }
+        } else {
+            throw new IllegalStateException("File " + fileName + " already exists in share " + share);
+        }
+    }
+
+    private void mkdirs(DiskShare share, String parentPath) {
+        if (parentPath == null || parentPath.isEmpty()) {
+            return;
+        }
+
+        String[] pathParts = parentPath.split("[/\\\\]");
+        String currentPath = "";
+
+        for (String part : pathParts) {
+            if (part.isEmpty()) {
+                continue;
+            }
+
+            currentPath = currentPath.isEmpty() ? part : currentPath + "/" + part;
+
+            if (!share.folderExists(currentPath)) {
+                share.mkdir(currentPath);
+            }
+        }
+    }
+}
```

Comment:
instead of throwing the ioexception in public methods, use try-catch and throw a runtimeexception please